aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -757,6 +860,31 @@ class AccumulationBuffer {
   at::DataPtr buffer_;
 };
 
+template <typename scalar_t>
+int get_output_vec_size(TensorIterator &iter) {
+  int vec_size = 4;
+  auto update_vec_size = [&vec_size](uint64_t n) {
+    while(n % vec_size != 0) {
+      vec_size /= 2;
+    }
+  };
+  
+  uint64_t base_address = reinterpret_cast<uint64_t>(iter.data_ptr(iter.noutputs())) / sizeof(scalar_t);
```

Comment:
hm, will this actually check the alignment of the first input rather than output?

Reply:
Reduction assumes only one input. And yes, it checks the address of input, because only load instructions are vectorized.